linux/iscsi.go

```diff
@@ -849,10 +850,16 @@ func iscsiGetTargetOfDevice(dev *model.Device) (target *model.IscsiTarget, err e
 			if err != nil {
 				return nil, fmt.Errorf("no iscsi target found from iscsi session %s", sessionID)
 			}
-			return iscsiTarget, nil
+			iscsiTargets = append(iscsiTargets, iscsiTarget)
 		}
 	}
-	return nil, nil
+	if iscsiTargets != nil {
```

Comment:
just return `iscsiTargets, nil` instead of checking the if condition

Reply:
The caller of this method would also need to be modified. 
Currently we set the device iscsiTarget but now we need to refer to iscsiTargets[0] atleast for non 3Pardata
```iscsiTarget, err := iscsiGetTargetOfDevice(device)
				if err != nil {
					log.Debugf("unable to get iscsi target for device %s. Error: %s", device.Pathname, err.Error())
					continue
					// continue with other devices, might be in offline state and IscsiTarget will be nil for this device
				}
				if iscsiTarget != nil {
					device.IscsiTarget = iscsiTarget
				}
			}
```